aten/src/ATen/core/jit_type.h

```diff
@@ -372,12 +390,13 @@ inline c10::optional<Stride> merge_primitive(
 
   auto merged_index =
       merge_primitive(left->stride_index_, right->stride_index_);
-  auto merged_cont = merge_primitive(left->contiguous_, right->contiguous_);
+  auto merged_cont =
```

Comment:
I think merge contiguity_type should be interleaved with the result of merged_index.

If stride index changed, very likely contiguity_type means different things now. so maybe we could have something like
`merged_cont = merged_index.has_value() ? NONCONTIGUOUS : merge_primitive(left->contiguity_type_, right->contiguity_type_);`